### PR TITLE
Fix documentation for round()

### PR DESCRIPTION
**Pull request by stephentgrammer**
Fix function signature and clarify documentation for `round()`.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -4882,14 +4882,11 @@ static if (real.mant_dig >= long.sizeof * 8)
...
 
 /*******************************************
- * Return the value of x rounded to the nearest integer.
- * If the fractional part of x is exactly 0.5, the return value is
- * rounded away from zero.
- *
  * Returns:
- *     A `real`.
+ *     A `real`, which is the value of x rounded to the nearest integer. If the fractional
+ *     part of x is exactly 0.5, the it is rounded away from zero.
```
Comment (ZombineDev) [inline std/math.d]: "...exactly 0.5, ~the~ it is rounded..."

**Comment by Hackerpilot** [comment]
Looks like the CI fails because if you use `real` instead of `auto` it can no longer determine that the function is `pure` for the posix-specific unittest.

**Comment by JackStouffer** [comment]
Yes, I changed it to auto so we didn't need to add two overloads.

@stephentgrammer in the future please avoid changes that don't add to the quality of code; it just represents churn.

**Comment by stephentgrammer** [comment]
Sounds good. Thanks for your work guys!

**Comment by Hackerpilot** [comment]
@JackStouffer If it weren't for the problem with the compiler failing to figure out `pure`, explicitly specifying the return type would be an improvement.

**Comment by JackStouffer** [comment]
Yes, but the implementation is what we got, and having `auto` in this case is the cleaner option.

**Pull request closed without merging**